migtests/scripts/functions.sh

```diff
@@ -708,4 +708,64 @@ validate_failure_reasoning() {
     fi
 }
 
+post_assess_migration() {
+    json_file="$EXPORT_DIR/assessment/reports/assessmentReport.json"
+    sharded_tables=$(fetch_sharded_tables "$json_file")
+    echo "Sharded Tables: $sharded_tables"
+    colocated_tables=$(fetch_colocated_tables "$json_file")
+    echo "Colocated Tables: $colocated_tables"
 
+    move_tables "$json_file" 30
+
+    updated_sharded_tables=$(fetch_sharded_tables "$json_file")
+    echo "Updated Sharded Tables: $updated_sharded_tables"
+    updated_colocated_tables=$(fetch_colocated_tables "$json_file")
+    echo "Updated Colocated Tables: $updated_colocated_tables"
+}
+
+fetch_sharded_tables() {
+    local json_file=$1
+    jq '.Sizing.SizingRecommendation.ShardedTables // []' "$json_file"
+}
+
+fetch_colocated_tables() {
+    local json_file=$1
+    jq '.Sizing.SizingRecommendation.ColocatedTables // []' "$json_file"
+}
+
+# Function to move a specified percentage of tables from colocated to sharded
+move_tables() {
+    local json_file=$1
+    local percentage=$2
```

Comment:
(Just a discussion, not a strong recommendation) 
I understand the intention behind randomizing the set of tables in colocated/sharded by percentags, but usually, I think it's better to have tests be deterministic, i.e. you should be able to reproduce a a test case failure by just running it. 
By depending on percentages, it makes it more complicated to do so. (although you could just modify it to 100% or 0% when we need to repro.

Alternatively, what if we just mark all tables are sharded? That way we get to test our DDL modification for all table types.

Reply:
@makalaaneesh  Regarding 100% Sharded tables, so that it modify all the tables is good from perspective of testing.

With that just my concern was that we should cover this general case(combination of colocated and sharded tables) with higher priority and not miss it

Reply:
Just a suggestion. We can make the percentage a random value as well (maybe from a specific set). That will ensure 0, 100 and anything in between.
Maybe choose from a set of 0, 25, 50, 75, 100?

Reply:
^ that would make it even more non-deterministic :D

Reply:
yeah makes sense to have a separate test script for schema migration